crates/pallet-grandpa-finality-verifier/src/lib.rs

```diff
@@ -124,16 +129,21 @@ pub mod pallet {
         chain_id: T::ChainId,
         init_params: InitializationData,
     ) -> DispatchResult {
-        let InitializationData { header, set_id } = init_params;
-        let header_decoded = C::decode_header::<T>(header.as_slice())?;
+        let InitializationData {
+            genesis_hash,
+            best_known_finalized_header,
+            set_id,
+        } = init_params;
+        let header_decoded = C::decode_header::<T>(best_known_finalized_header.as_slice())?;
         let change =
             find_scheduled_change(&header_decoded).ok_or(Error::<T>::UnsupportedScheduledChange)?;
         // Set the best known finalized header
-        BestKnownFinalized::<T>::insert(chain_id, header);
+        BestKnownFinalized::<T>::insert(chain_id, best_known_finalized_header);
         let authority_set = AuthoritySet {
             authorities: change.next_authorities,
             set_id,
         };
+        LastImportedBlock::<T>::insert(chain_id, (0, genesis_hash));
```

Comment:
So we are not going to store the genesis block itself, right? Should be fine for Substrate-based chains.